libres/lib/analysis/update.cpp

```diff
@@ -668,9 +670,13 @@ void copy_parameters(enkf_fs_type *source_fs, enkf_fs_type *target_fs,
         stringlist_free(param_keys);
     }
 }
+
+using node =
+    std::unique_ptr<enkf_node_type, std::function<void(enkf_node_type *)>>;
```

Comment:
`std::function` is a particularly inefficient implementation of a closure, so I'd advise against using it willy-nilly (as in, most often a function pointer will suffice).

There's only one way to destruct a `enkf_node_type` and that is via `enkf_node_free__`, so I'd write the following:

```suggestion
const auto enkf_node_deleter = [](enkf_node_type* ptr) { enkf_node_free(ptr); };

using node =
    std::unique_ptr<enkf_node_type, decltype(enkf_node_deleter)>;
```

Reply:
> There's only one way to destruct a enkf_node_type and that is via enkf_node_free__

Ehhh - no, the slightly better way in this case is to call `enkf_node_free()` which takes a pointer of the right type and not `void *`